### PR TITLE
Solution: JavaScript I

### PR DESCRIPTION
**Pull request by nickcannariato**
### Objects
To better understand objects, you really just need to write more of them. The [objects.js](assignments/objects.js) file contains several challenges centered around a theme of interns starting at a new job. The Human Resources team needs information about the new hires. Use your new found object skills answer vital questions for HR.

* Read the instructions found within the file carefully to finish the challenges. 
* Complete each challenge presented before moving on to Arrays.

### Arrays
The [arrays.js](assignments/arrays.js) assignment takes us through a large data set of used cars.  You have been asked to help a used car business with some customer requests based on their inventory.  Use for loops and arrays to solve their problems.

* Utilize the the array `inventory` to complete your challenges
* You are not permitted to use map, reduce, or filter to solve these problems.  Only use a basic for loop.
* Complete each challenge presented before moving on to stretch.

### Stretch

* [x] Arrow Function Syntax - [Check out this awesome guide for ES6 arrow syntax](https://medium.freecodecamp.org/when-and-why-you-should-use-es6-arrow-functions-and-when-you-shouldnt-3d851d7f0b26). You will see more and more arrow functions as you progress deeper into JavaScript. Use the [stretch-function-conversion.js](assignments/stretch-function-conversion.js) file as a helper challenge to showcase some of the differences between ES5 and ES6 syntax.

### CODE REVIEW

File: assignments/arrays.js [hunk 1]
```diff
@@ -63,34 +63,52 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
...
+let lastCar = {...inventory[inventory.length - 1]};
+console.log(lastCar.car_make, lastCar.car_model);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
+let carModels = inventory.sort((a, b) => {
+  return a.car_model.toLowerCase() > b.car_model.toLowerCase() ? 1 : b.car_model.toLowerCase() > a.car_model.toLowerCase() ? -1 : 0;
+});
```
Comment (sabrinagear) [inline assignments/arrays.js]: Excellent attention to detail here, Nick! Awesome job accounting for the fact that the sort() method uses a compare function that defines the sort order. If omitted, the array is sorted according to each character's Unicode code point value, according to the string conversion of each element. Thus, lowercase letters are sorted after upper-case letters. Nice implementation here.

File: assignments/arrays.js [hunk 2]
```diff
@@ -63,34 +63,52 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
...
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
-
+let car33 = {};
+for (let i = 0; i < inventory.length; i++) {
+  if (inventory[i].id === 33) {
+    car33 = {...inventory[i]};
+  };
+}
```
Comment (sabrinagear) [inline assignments/arrays.js]: Nice job. Interesting use of the spread operator here. If you wanted to define a variable to identify the car's id that you are searching for, a simpler method would then be to use find(): 

```suggestion
let id = 33;
const car = inventory.find(car => car.id === id);
console.log(`Car ${id} is a ${car.car_year} ${car.car_make} ${car.car_model}`);
```

File: assignments/arrays.js [hunk 3]
```diff
@@ -63,34 +63,52 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
...
+console.log(carYears);
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars =[];
-console.log(); 
+for (let i = 0; i < carYears.length; i++) {
+  if (carYears[i] < 2000) oldCars.push(carYears[i]);
+}
```
Comment (sabrinagear) [inline assignments/arrays.js]: great job!

File: assignments/arrays.js [hunk 4]
```diff
@@ -63,34 +63,52 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
...
 
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi =[];
-console.log();
+for (let i = 0; i < inventory.length; i++) {
+  if (inventory[i].car_make.toLowerCase() === 'bmw' || inventory[i].car_make.toLowerCase() === 'audi') {
+    BMWAndAudi.push(inventory[i]);
+  };
+}
```
Comment (sabrinagear) [inline assignments/arrays.js]: Nice job accounting for case-sensitivity.

File: assignments/objects.js [hunk 5]
```diff
@@ -18,27 +18,70 @@ const example = {
...
 // Gannie's name
+console.log(`${gannie.name}'s name: ${gannie.name}`);
 
 // Antonietta's Gender
+console.log(`${antonietta.name}'s gender: ${antonietta.id}`);
 
 // ==== Challenge 3: Object Methods ==== 
 // Give Kennan the ability to say "Hello, my name is Kennan!" Use the console.log provided as a hint.
 // console.log(kennan.speak());
+console.log(kennan.speak());
```
Comment (sabrinagear) [inline assignments/objects.js]: Can you think of a way to assign a method from outside of the object literal?

File: assignments/stretch-function-conversion.js [hunk 6]
```diff
@@ -1,23 +1,28 @@
...
+let subtract = (param1, param2) => param1 - param2;
 
 // exampleArray = [1,2,3,4];
 // const triple = exampleArray.map(function (num) {
 //   return num * 3;
 // });
-// console.log(triple);
\ No newline at end of file
+// console.log(triple);
+const triple = exampleArray.map(num => num * 3);
```
Comment (sabrinagear) [inline assignments/stretch-function-conversion.js]: excellent!

File: assignments/arrays.js (hunk 2 above)
Comment (nickcannariato) [reply assignments/arrays.js]: Ah, good call on using `.find()` instead. Thanks!

**Pull request merged**